infinite-tracing/src/main/java/com/newrelic/ResponseObserver.java

```diff
@@ -96,4 +101,42 @@ private boolean isOkHttpALPNException(RuntimeException cause) {
         // It's the only mechanism we have to identify the problem.
         return cause.getMessage() != null && cause.getMessage().startsWith("TLS ALPN negotiation failed with protocols");
     }
+
+    /**
+     * Handle the error {@code status}. If irrecoverable, close the connection. Else, backoff and initiate reconnection.
+     *
+     * @param status the error status
+     */
+    void handleErrorStatus(Status status) {
```

Comment:
The previous setup with a `BackoffPolicy` interface, implemented by `ConnectBackoffPolicy` and `DefaultBackoffPolicy`, and managed by `DisconnectionHandler` had poor "power to weight ratio". It resulted in the relatively simple workflow of handling status errors being distributed across 5 classes, and hard to put together. 

The `ResponseObserver#onError()` already does most of the error management, so why not include this last little bit so that its all in one spot?